### PR TITLE
Core/Creatures: Added possibility to automatically despawn personal s…

### PR DESCRIPTION
**Pull request by mdX7**
…pawns on quest remove in creature_summoned_data

**Tests performed:**
tested ingame, with #29090 merged and
```sql
DELETE FROM `creature_summoned_data` WHERE `CreatureID` IN(166787, 166792, 166797, 166782, 166784, 166800, 151088, 151089, 154170, 156626, 149917, 156622, 198896, 198883);
INSERT INTO `creature_summoned_data` (`CreatureID`, `QuestsToDespawnOnQuestRemove`) VALUES
-- Exiles Reach Emergency First Aid,Finding the Lost Expedition
(166787, '59930,59931'), -- Bo Standing
(166792, '59930,59931'), -- Mithdran Standing
(166797, '59930,59931'), -- Lana Jordan Standing
(166782, '59930'), -- Warlord Breka Grimaxe Beach
(166784, '59930'), -- Throg Beach
(166800, '59930'), -- Jinhake Beach
(151088, '54951,54952'), -- Kee La Standing
(151089, '54951,54952'), -- Bjorn Stouthands Standing
(154170, '54951,54952'), -- Austin Huxworth Standing
(156626, '54951'), -- Captain Garrick Standing
(149917, '54951'), -- Cole Beach
(156622, '54951'), -- Richter Beach
-- An Unlikely Informant / The New Clessington Estate
(198896, '72405,72426'), -- Mathias Shaw
(198883, '72405,72426'); -- Vanessa VanCleef
```

### CODE REVIEW

**Comment by Ovahlord** [comment]
usually personal spawns that are involved in quests have a summon aura which should be tied to them. especially in older xpacs there are dummy auras used for this behavior

**Comment by malcrom** [comment]
was ist das?

**Comment by mdX7** [comment]
> usually personal spawns that are involved in quests have a summon aura which should be tied to them. especially in older xpacs there are dummy auras used for this behavior

Usually, yes. But not all of them, so we could cover those without auras with this magic

**Comment by malcrom** [comment]
```
151088, '59930'), -- Kee La Standing
```
should be

```
151088, '54951'), -- Kee La Standing
```

File: src/server/game/Globals/ObjectMgr.cpp [hunk 1]
```diff
@@ -733,6 +733,19 @@ void ObjectMgr::LoadCreatureSummonedData()
...
         }
 
+        if (!fields[4].IsNull())
+        {
+            for (std::string_view questStr : Trinity::Tokenize(fields[4].GetStringView(), ',', false))
+            {
+                if (Optional<uint32> questId = Trinity::StringTo<uint32>(questStr))
+                {
+                    // quests aren't loaded yet, we cannot validate them
+                    summonedData.QuestsToDespawnOnQuestRemove.emplace();
```
Comment (Traesh) [inline src/server/game/Globals/ObjectMgr.cpp:743]: This should be done before the for loop (emplace destroy already existing value if any)

**Pull request merged**